.github/workflows/mypy.yml

```diff
@@ -0,0 +1,14 @@
+name: "Patch typing check"
+
+on:
+  push:
+    branches:
+      - main
+      - staging
+  pull_request:
+  merge_group:
+
+jobs:
+  patch-typing-check:
+    name: Run Patch Type Check
+    uses: codecov/gha-workflows/.github/workflows/mypy.yml@may_06_mypy
```

Comment:
TODO: change the pinned version once the gha-workflows PR is merged and a new release is created